src/msw/webview_ie.cpp

```diff
@@ -853,25 +857,229 @@ wxString wxWebViewIE::GetPageText() const
+                return wxWEBVIEWIE_EMULATION_LEVEL_11_DEFAULT;
+            case wxWEBVIEWIE_EMULATION_LEVEL_10:
+                return wxWEBVIEWIE_EMULATION_LEVEL_10;
+            case wxWEBVIEWIE_EMULATION_LEVEL_10_DEFAULT:
+                return wxWEBVIEWIE_EMULATION_LEVEL_10_DEFAULT;
+            case wxWEBVIEWIE_EMULATION_LEVEL_9:
+                return wxWEBVIEWIE_EMULATION_LEVEL_9;
+            case wxWEBVIEWIE_EMULATION_LEVEL_9_DEFAULT:
+                return wxWEBVIEWIE_EMULATION_LEVEL_9_DEFAULT;
+            case wxWEBVIEWIE_EMULATION_LEVEL_8:
+                return wxWEBVIEWIE_EMULATION_LEVEL_8;
+            case wxWEBVIEWIE_EMULATION_LEVEL_8_DEFAULT:
+                return wxWEBVIEWIE_EMULATION_LEVEL_8_DEFAULT;
+            case wxWEBVIEWIE_EMULATION_LEVEL_7_DEFAULT:
+                return wxWEBVIEWIE_EMULATION_LEVEL_7_DEFAULT;
+            default:
+                return wxWEBVIEWIE_EMULATION_LEVEL_EMPTY;
+        }
+    }
+    wxLogWarning(_("Registry key SOFTWARE\\Microsoft\\Internet Explorer\\Main\\FeatureControl\\FEATURE_BROWSER_EMULATION \
+        does not exist"));
+    return wxWEBVIEWIE_EMULATION_LEVEL_EMPTY;
+}
+
+bool wxWebViewIE::SetEmulationLevel(wxWebViewIEEmulationLevel level)
+{
+    wxRegKey key(wxRegKey::HKCU, _T("SOFTWARE\\Microsoft\\Internet Explorer\\Main\\FeatureControl\\FEATURE_BROWSER_EMULATION"));
+    if ( key.Exists() )
+    {
+        if ( !key.SetValue(wxGetFullModuleName().AfterLast('\\'), level) )
+        {
+            wxLogWarning(_("Can not set level value of registry key \
+                SOFTWARE\\Microsoft\\Internet Explorer\\Main\\FeatureControl\\FEATURE_BROWSER_EMULATION"));
+            return false;
+        }
+        return true;
+    }
+    wxLogWarning(_("Registry key SOFTWARE\\Microsoft\\Internet Explorer\\Main\\FeatureControl\\FEATURE_BROWSER_EMULATION \
+        does not exist"));
+    return false;
+}
+
+bool wxWebViewIE::RunScript(const wxString& javascript, wxString* output)
 {
     wxCOMPtr<IHTMLDocument2> document(GetDocument());
+    IDispatch* scriptDispatch = NULL;
 
-    if(document)
+    if ( !document )
     {
-        wxCOMPtr<IHTMLWindow2> window;
-        wxString language = "javascript";
-        HRESULT hr = document->get_parentWindow(&window);
-        if(SUCCEEDED(hr))
+        wxLogWarning(_("HTML document is null"));
+        return false;
+    }
+
+    if ( FAILED(document->get_Script(&scriptDispatch)) )
+    {
+        wxLogWarning(_("Can't get the Javascript"));
+        return false;
+    }
+
+    wxString counter;
+    wxString javaScriptVariable =
+        wxWebViewUtils::WrapJavaScript(javascript,
+        &m_runScriptCount, &counter);
+
+    wxAutomationObject scriptAO(scriptDispatch);
+    wxVariant varJavascript(javaScriptVariable);
+    wxVariant varResult;
+
+    if ( !scriptAO.Invoke("eval", DISPATCH_METHOD, varResult, 1, &varJavascript) )
+    {
+        wxLogWarning(_("Can't run Javascript"));
```

Comment:
This message should also be part of the helper function instead of being repeated multiple times.